### PR TITLE
[DOCS] Add release notes for 7.17.1

### PR DESCRIPTION
**Pull request by pugnascotia**
https://elasticsearch_84275.docs-preview.app.elstc.co/diff

### CODE REVIEW

**Review by jrodewig — APPROVED** [review]
LGTM. Thanks!

I'm going to request a review from the ML writers for the ML-related sections.

File: docs/reference/release-notes/7.17.asciidoc [hunk 1]
```diff
@@ -1,3 +1,103 @@
...
+Ingest::
+* Fix `GeoIpDownloader` startup during rolling upgrade {es-pull}84000[#84000]
+* Short circuit date patterns after first match {es-pull}83764[#83764] (issue: {es-issue}73918[#73918])
+
+Machine Learning::
+* Fix bug where initial scale from 0->1 could scale too high {es-pull}84244[#84244]
+* Retry anomaly detection job recovery during relocation {es-pull}83456[#83456]
+* Address cause of "[CStatisticalTests.cc@102] Test statistic is nan" log errors {ml-pull}2196[#2196]
+* Address possible causes of "x = NaN, distribution = N5boost4math23students_t_distribution" log errors {ml-pull}2197[#2197]
+* Fix bug restoring data gatherer state for time of day and week anomaly detection functions. This could lead to "No queue item for time " and "Time is out of range. Returning earliest bucket index" log errors  {ml-pull}2213[#2213]
```
Comment (lcawl) [inline docs/reference/release-notes/7.17.asciidoc:71]: Shortening long description:

```suggestion
* Fix bug restoring data gatherer state for time of day and week anomaly detection functions  {ml-pull}2213[#2213]
```

**Review by lcawl — APPROVED** [review]
MLR PRs LGTM, thanks!

**Comment by pugnascotia** [comment]
Thanks all!

**Pull request merged**